wdl/src/main/scala/wdl/Declaration.scala

```diff
@@ -127,11 +136,17 @@ object Declaration {
       } yield GraphOutputDeclarationNode(graphOutputNode)
     }
 
-    decl match {
-      case Declaration(opt: WomOptionalType, _, None, _, _) => Valid(InputDeclarationNode(OptionalGraphInputNode(decl.womIdentifier, opt)))
-      case Declaration(_, _, None, _, _) => Valid(InputDeclarationNode(RequiredGraphInputNode(decl.womIdentifier, decl.womType)))
-      case Declaration(_, _, Some(expr), _, _) => declarationAsExpressionNode(expr)
-      case WorkflowOutput(_, _, expr, _, _) => workflowOutputAsGraphOutputNode(expr)
+    def asWorkflowInput(inputDefinition: InputDefinition): GraphInputNode = inputDefinition match {
+      case RequiredInputDefinition(_, womType) => RequiredGraphInputNode(decl.womIdentifier, womType)
+      case OptionalInputDefinition(_, optionalType) => OptionalGraphInputNode(decl.womIdentifier, optionalType)
+      case InputDefinitionWithDefault(_, womType, default) => OptionalGraphInputNodeWithDefault(decl.womIdentifier, womType, default)
```

Comment:
I'm not sure exactly what an `InputDefinitionWithDefault` is. Is this only optionals with a default, which are overrideable? Or does that also include non-optionals and those are overrideable too?

Reply:
Maybe a rename would help? It's a workflow input that has a default. Could be`Int x = 5`, could be `String? s = "hello"`

Reply:
I don't have a suggestion for a better name, as long as the doc on this thing says what you just said that should be fine. 🙂

Reply:
Added scaladocs to the `case class`